native/more/relationship-list.react.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import type { NavigationRoute } from '../navigation/route-names';
+import type { AppState } from '../redux/redux-setup';
+import type { MoreNavigationProp } from './more.react';
+
+import * as React from 'react';
+import { View, Text, FlatList } from 'react-native';
+
+import { connect } from 'lib/utils/redux-utils';
+import { type UserInfo } from 'lib/types/user-types';
+
+import { AddFriendsModalRouteName } from '../navigation/route-names';
+import { styleSelector } from '../themes/colors';
+
+import RelationshipListItem from './relationship-list-item.react';
+
+const DATA: UserInfo[] = [
+  { id: '1', username: 'John' },
+  { id: '2', username: 'Emma' },
+];
+
+type ListItem =
+  | {| type: 'header' |}
+  | {| type: 'footer' |}
+  | {| type: 'user', userInfo: UserInfo, lastListItem: boolean |};
+
+type Props = {|
+  navigation: MoreNavigationProp<>,
+  route: NavigationRoute<'FriendList' | 'BlockList'>,
+  // Redux state
+  styles: typeof styles,
+|};
+class RelationshipList extends React.PureComponent<Props> {
+  render() {
+    const listData = this.mapListData();
+
+    const empty = (
+      <Text
+        style={this.props.styles.emptyText}
+      >{`You haven't added any users yet`}</Text>
+    );
+
+    return (
+      <View style={this.props.styles.container}>
+        <FlatList
+          contentContainerStyle={this.props.styles.contentContainer}
+          ListEmptyComponent={empty}
+          data={listData}
+          renderItem={this.renderItem}
+        />
+      </View>
+    );
+  }
+
+  mapListData = (): ListItem[] => {
+    const users = DATA;
```

Comment:
This should be memoized based on the input data, so that it doesn't get recomputed every render. Since the memoization will be done using `reselect` long-term and there's no way to mock fake input to `reselect`, I guess we can do this later...

An alternative would be to make your mocked `DATA` array just be what you pass to `FlatList`. This would mean shelving the `mapListData` code and bringing it back later once you have the actual data in place.

Reply:
Done. But I'm a little bit confused. I was trying to do it the same way it is done on [thread-settings](https://github.com/Ashoat/squadcal/blob/master/native/chat/settings/thread-settings.react.js#L318). And looks like `listData` in there is recalculated every render since it happens directly on render method. Even if the selector used in `getThreadInfo` is memoized 🤔